docs/book/src/tasks/experimental-features/cluster-class/change-clusterclass.md

```diff
@@ -1,32 +1,192 @@
+- [Dry running ClusterClass changes](#Dry running ClusterClass changes) before applying them.
+
+If instead you are interested in understanding more about how which kind of  
+effects you should expect on the Clusters, or if you are interested in additional details 
+about the internals of the topology reconciler you can start reading the notes in the 
+[dry run](#Dry running ClusterClass changes) documentation or looking at the [reference](#Reference)
+documentation at the end of this page.
+
+## Changing ClusterClass templates
+
+Templates are an integral part of a Cluster Class, and thus the same considerations
+described in the previous paragraph applies; that's means that also when you are changing
+a template referenced in a ClusterClass users should always plan for how the
+change should be propagated to the existing Clusters and choose the strategy that best
+suite the expectation.
+
+Then, according to the [Cluster API operational practices](https://cluster-api.sigs.k8s.io/tasks/updating-machine-templates.html),
+the recommended way for updating templates is by template rotation:
+- create a new template
+- update the template reference in the ClusterClass
+- then delete the old template
+
+<aside class="note">
+<h1>In place template mutations</h1>
+
+In case a provider supports in place template mutations, the Cluster API topology controller 
+will adapt to them at the next reconciliation, but the system is not watching for those 
+specific changes. That means that when the underlying template is updated in this way the changes 
+may not be reflected immediately, but it will be put in place at the next full reconciliation. 
+The maximum time for the next reconciliation to take place is related to the CAPI controller
+sync period - 10 minutes by default.
+
+</aside>
 
 <aside class="note warning">
+<h1>Reusing templates across ClusterClasses</h1>
+
+As already discussed in [writing a cluster class](write-clusterclass.md), technically it is possible to 
+re-use a template across ClusterClasses, but this practice is not recommended because it makes it more
+complex to reason about the impact that any change on such a template can have on existing Clusters.
+
+</aside>
+
+Also in case of changes to the ClusterClass templates, please make sure to:
+
+- Understand what [Compatibility Checks](#Compatibility Checks) are and how to prevent changes
+  that can lead to non-functional Clusters.
+- [Dry running ClusterClass changes](#Dry running ClusterClass changes) before applying them.
+
+And eventually learn more about this reading the notes in the [dry run](#Dry running ClusterClass changes) documentation or
+looking at the [referece](#Reference) documentation at the end of this page.
+
+## Rebase
 
+Rebasing is an operational practice for transitioning a Cluster from one ClusterClass to another,
+and the operation can be triggered by simply changing the value in `cluster.spec.topology.class`.
+
+Also in this case, please make sure to:
```

Comment:
This reads a bit repetitive as we have (if I saw correctly) the same block 3 times. 
If you see a good way to deduplicate or otherwise mitigate the issue, it would be nice.

Otherwise just leave it as is, we probably can't emphasize that point enough :)

Reply:
The first instance of this sentence is more detailed, the next two repetitions are shortened, but the intent was to stress the relevance of that point

Reply:
Absolutely fine.